LoginServer/LoginServerForm.cs

```diff
@@ -0,0 +1,200 @@
+
+            MessageLogTB.Text += str + "\r\n";
+        }
+
+        /// <summary>
+        /// Display the received message.
+        /// </summary>
+        /// <param name="message">The message received.</param>
+        private void DisplayMessageReceived(byte[] message)
+        {
+            string str = Utility.ConvertToString(message);
+
+            MessageLogTB.Text += str + "\r\n";
+        }
+        #endregion
+
+        // ClientDisconnected() & NoClients().
+        #region ClientDisconnectedMethods
+        /// <summary>
+        /// Prints the number of connected clients everytime one connects or disconnects.
+        /// Calls the NoClients() when there are no clients.
+        /// </summary>
+        private void ClientDisconnected()
+        {
+            string str = "Total Clients: " + pipeServer.TotalConnectedClients;
+            MessageLogTB.Text += str + "\r\n";
+
+            if (pipeServer.TotalConnectedClients == 0)
+            {
+                NoClients();
+            }
+        }
+
+        /// <summary>
+        /// This method dissables the messaging functions when there is no clients connected.
+        /// </summary>
+        private void NoClients()
+        {
+            SendBtn.Enabled = false;
+            SendMessageTB.Enabled = false;
+            ClearBtn.Enabled = false;
+        }
+        #endregion
+
+        #region ButtonClickMethods
+
+        /// <summary>
+        /// Starts the server.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void StartBtn_Click(object sender, EventArgs e)
+        {
+            pipeServer.Start(PipeNameTB.Text);
+            StartBtn.Enabled = false;
+            MessageLogTB.Text += "The Server is Now Running: \r\n";
+        }
+
+        /// <summary>
+        /// Send the text in the send message box to all the clients.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void SendBtn_Click(object sender, EventArgs e)
+        {
+            if (!String.IsNullOrWhiteSpace(SendMessageTB.Text))
+            {
+                byte[] messageBuffer = Utility.ConvertToBytes(SendMessageTB.Text);
+
+                pipeServer.SendMessageToAll(messageBuffer);
+            }
+            SendMessageTB.Clear();
+            SendMessageTB.Focus();
+        }
+
+        /// <summary>
+        /// Clears the message log.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void ClearBtn_Click(object sender, EventArgs e)
+        {
+            MessageLogTB.Clear();
+            SendMessageTB.Focus();
+        }
+        #endregion
+
+        private void LoginServerForm_FormClosing(object sender, FormClosingEventArgs e)
+        {
+            if (pipeServer != null)
+            {
+                if (pipeServer.getListenThread() != null)
```

Comment:
Oddness.
You have a pipeServer.Start.
Suggest moving this into pipeServer.Stop
for consistency/clarity